src/parser.rs

```diff
@@ -166,51 +155,244 @@ impl<'a> DocParser<'a> {
+                &file_dep.specifier,
+                first_def.module.specifier(),
+              )?;
+              return self.resolve_dangling_reference(
+                specifier,
+                reference_def,
+                name_path,
+                true,
+              );
+            }
+            DefinitionKind::Definition => {
+              for (i, name_path_item) in name_path.iter().enumerate() {
+                let Some((name, id)) = export_symbol
+                  .exports()
+                  .iter()
+                  .find(|(name, _)| &name_path_item == name)
+                else {
+                  let definitions = self
+                    .root_symbol
+                    .go_to_definitions(export.module, export_symbol)
+                    .collect::<Vec<_>>();
+
+                  return self.resolve_dangling_reference(
+                    definitions.first().unwrap().module.specifier(),
+                    reference_def,
+                    // -1 to include the root
+                    name_path[i - 1..].to_vec(),
+                    false,
+                  );
+                };
+
+                export_name = name;
+                export_symbol = export.module.symbol(*id).unwrap();
+              }
+
+              definitions = self
+                .root_symbol
+                .go_to_definitions(export.module, export_symbol)
+                .collect::<Vec<_>>();
+            }
+          }
+        }
+
+        let original_range = &export_symbol.decls().first().unwrap().range;
+
+        if let Some(first_def) = definitions.first() {
+          use deno_graph::symbols::DefinitionKind;
+          match first_def.kind {
+            DefinitionKind::ExportStar(_) => {}
+            DefinitionKind::Definition => {
+              if star || first_def.module.specifier() != module_info.specifier()
+              {
+                for definition in &definitions {
+                  if definition_location(definition) == reference_def.target {
+                    let decl = definition.symbol_decl;
+                    let mut maybe_docs = self.docs_for_maybe_node(
+                      definition.module,
+                      definition.symbol,
+                      decl.maybe_node(),
+                      module_info.esm(),
+                      first_def.module.specifier(),
+                      Some(decl),
+                      Some(original_range),
+                    );
+                    if !maybe_docs.is_empty() {
+                      for doc_node in &mut maybe_docs {
+                        doc_node.name = export_name.as_str().into();
+                        doc_node.declaration_kind = DeclarationKind::Export;
+                      }
+
+                      return Ok(Some(maybe_docs));
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+
+        unreachable!()
```

Comment:
Seems risky.

Reply:
changed to debug assertion only and returns none on prod. though this really shouldnt hit